src/Parser/Core/Modules/Features/BaseHealerStatWeights.js

```diff
@@ -0,0 +1,373 @@
+        this._handleHealEvent(event, healVal);
+      }
+    }
+  }
+  _handleHealEvent(event, healVal) {
+    const spellInfo = this._getSpellInfo(event);
+    const targetHealthPercentage = (event.hitPoints - healVal.effective) / event.maxHitPoints; // hitPoints contains HP *after* the heal
+    this._handleHeal(spellInfo, event, healVal, targetHealthPercentage);
+  }
+  /**
+   * This actually does that stat weights calculations.
+   * @param spellInfo An object containing information about how to treat the spell, like on what stats it scales.
+   * @param eventForWeights The event that uses the stats to scale its healing. If we're tracking a multiplier (like Beacons), this should be the source of the multiplier.
+   * @param {HealingValue} healVal The actual healing done, all child weight calculators should use this instead of the event data as that might be for another event.
+   * @param targetHealthPercentage The percentage of health the target has remaining BEFORE the heal.
+   * @private
+   */
+  _handleHeal(spellInfo, eventForWeights, healVal, targetHealthPercentage) {
+    // Most spells are counted in healing total, but some spells scale not on their own but 'second hand' from other spells
+    // I adjust them out of total healing to preserve some accuracy in the "Rating per 1%" stat.
+    // Good examples of multiplier spells are Leech and Velens.
+    if (!spellInfo.multiplier) {
+      this.totalAdjustedHealing += this._adjustGain(healVal.effective, targetHealthPercentage);
+    }
+
+    if (spellInfo.ignored) {
+      return;
+    }
+
+    this.totalOneLeech += this._adjustGain(this._leech(eventForWeights, healVal), targetHealthPercentage);
+
+    if (spellInfo.multiplier) {
+      // Multiplier spells aren't counted for weights because they don't **directly** benefit from stat weights
+      return;
+    }
+
+    if (spellInfo.int) {
+      this.totalOneInt += this._adjustGain(this._intellect(eventForWeights, healVal), targetHealthPercentage);
+    }
+    if (spellInfo.crit) {
+      this.totalOneCrit += this._adjustGain(this._criticalStrike(eventForWeights, healVal), targetHealthPercentage);
+    }
+    if (spellInfo.hasteHpct) {
+      this.totalOneHasteHpct += this._adjustGain(this._hasteHpct(eventForWeights, healVal), targetHealthPercentage);
+    }
+    if (spellInfo.mastery) {
+      this.totalOneMastery += this._adjustGain(this._mastery(eventForWeights, healVal), targetHealthPercentage);
+    }
+    if (spellInfo.vers) {
+      this.totalOneVers += this._adjustGain(this._versatility(eventForWeights, healVal), targetHealthPercentage);
+    }
+  }
+  _adjustGain(gain, targetHealthPercentage) {
+    // We want 0-20% health to get value gain, and then linearly decay to 100% health
+    const maxValueHealthPercentage = 0.3;
+    const mult = 1 - Math.max(0, (targetHealthPercentage - maxValueHealthPercentage) / (1 - maxValueHealthPercentage));
+    return this.scaleWeightsWithHealth ? gain * mult : gain;
```

Comment:
Maybe just use `this.scaleWeightsWithHealth` to do an early return as first line? Doesn't matter that much though.

Reply:
I don't want to suggest I intended there to be multiple scalers mixed